controllers/activemqartemis_controller_test.go

```diff
@@ -7208,6 +7207,91 @@ var _ = Describe("artemis controller", func() {
 			// cleanup
 			CleanResource(&crd, crd.Name, defaultNamespace)
 		})
+
+		It("expect error applying strategic merge patch", func() {
+
+			By("By creating a crd with template")
+			ctx := context.Background()
+			crd := generateArtemisSpec(defaultNamespace)
+
+			crd.Spec.ResourceTemplates = []brokerv1beta1.ResourceTemplate{
+				{
+					Selector:    &brokerv1beta1.ResourceSelector{Kind: ptr.To("StatefulSet")},
+					Annotations: map[string]string{"someSsKey": "someSsValue"},
+					Patch: &unstructured.Unstructured{
+						Object: map[string]interface{}{
+							"kind": "StatefulSet",
+							"spec": map[string]interface{}{
+								"template": map[string]interface{}{
+									"spec": map[string]interface{}{
+										"containers": "INVALID_VALUE",
+									},
+								},
+							},
+						},
+					},
+				},
+			}
+
+			Expect(k8sClient.Create(ctx, &crd)).Should(Succeed())
+
+			createdCrd := &brokerv1beta1.ActiveMQArtemis{}
+			brokerKey := types.NamespacedName{Name: crd.Name, Namespace: crd.Namespace}
+
+			By("verifying started via Status")
+			Eventually(func(g Gomega) {
+				g.Expect(k8sClient.Get(ctx, brokerKey, createdCrd)).Should(Succeed())
+				deployedCondition := meta.FindStatusCondition(createdCrd.Status.Conditions, brokerv1beta1.DeployedConditionType)
+				g.Expect(deployedCondition).NotTo(BeNil())
+				g.Expect(deployedCondition.Status).To(Equal(metav1.ConditionFalse))
+				g.Expect(deployedCondition.Reason).To(Equal(brokerv1beta1.DeployedConditionCrudKindErrorReason))
+				g.Expect(deployedCondition.Message).To(ContainSubstring("error applying strategic merge patch"))
+			}, existingClusterTimeout, existingClusterInterval).Should(Succeed())
+
+			// cleanup
+			CleanResource(&crd, crd.Name, defaultNamespace)
+		})
+
+		It("expect failed to create new StatefulSet", func() {
```

Comment:
I would also rename this one to something more specific:

`It("should report validation error when patched resource contains invalid field values", func() {`

And probably a comment later on explaining why the INVALID_VALUE isn't compatible with the enum of the podManagementPolicy

Reply:
I'll use a more specific description.